pandas/indexes/multi.py

```diff
@@ -656,9 +665,74 @@ def _has_complex_internals(self):
         return True
 
     @cache_readonly
+    def is_monotonic(self):
+
+        def level_values(level):
+            unique = self.levels[level]
+            labels = self.labels[level]
+            return algos.take_1d(unique.values, labels,
+                                 fill_value=unique._na_value)
```

Comment:
Can you have null values in the uniques here?  If the level values contain NaNs, then I imagine the behavior of `lexsort` is undefined, which means you could get false positives.